backend/executor.go

```diff
@@ -181,9 +189,27 @@ func (executor *grpcExecutor) ExecuteActivity(ctx context.Context, iid api.Insta
 }
 
 // Shutdown implements Executor
-func (g *grpcExecutor) Shutdown(ctx context.Context) {
+func (g *grpcExecutor) Shutdown(ctx context.Context) error {
 	// closing the work item queue is a signal for shutdown
 	close(g.workItemQueue)
+
+	// Iterate through all pending items and close them to unblock the goroutines waiting on this
```

Comment:
Are these changes to the `Shutdown` routine related to the original issue or is this unrelated?

Reply:
This was my first attempt at fixing the issue. It turned out it didn't fix it, but it seemed like something that would be helpful to have nevertheless, to ensure we don't leave goroutines waiting

Reply:
Should we not delete the items as well from pendingActivities/pendingOrchestrators as well in case of shutdown?

Reply:
The idea is that shutdown is called before the object is abandoned. The memory is released by the GC regardless. The important thing is to close the channels in case there are goroutines waiting on them.